presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestTimeUuidFunctions.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.cassandra;
+
+import io.prestosql.operator.scalar.AbstractTestFunctions;
+import io.prestosql.spi.type.SqlTimestampWithTimeZone;
+import io.prestosql.spi.type.TimeZoneKey;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.ZonedDateTime;
+
+import static io.prestosql.spi.type.TimestampWithTimeZoneType.createTimestampWithTimeZoneType;
+import static io.prestosql.spi.type.VarcharType.createVarcharType;
+import static java.time.ZoneOffset.UTC;
+
+public class TestTimeUuidFunctions
+        extends AbstractTestFunctions
+{
+    @BeforeClass
+    protected void registerFunctions()
+    {
+        functionAssertions.installPlugin(new CassandraPlugin());
+    }
+
+    @Test
+    public void testMinTimeuuid()
+    {
+        assertFunction("min_timeuuid(TIMESTAMP '2013-01-01 00:05:00.000')", createVarcharType(36), "89816e00-5331-11e2-8080-808080808080");
```

Comment:
I'm confused. We're testing with `TIMESTAMP`, but the function is defined for `TIMESTAMP WITH TIME ZONE`. This works because of the implicit coercion between those types, but my question is what's the main use case for this. If it's intended to be used just with `TIMESTAMP`, then we should change the function definitions.